### PR TITLE
Use node 18

### PR DESCRIPTION
**Pull request by brad-decker**
## Explanation
Upgrades minimum node version to v18.17. Version 18 is a requirement for Yarn version 4. See #20249 for more information. 

## Manual Testing Steps

There should be no functional changes in this PR

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -559,7 +559,7 @@
     "yargs": "^17.0.1"
   },
   "engines": {
-    "node": "^16.20.0",
+    "node": "^18.17.0",
```
Comment (Gudahtt) [inline package.json:562]: Two questions:

* any reason to set v18.17 as a minimum?
* Thoughts on using `>=` instead to allow v20 to be used as well? Or using something like `^18 || ^20`

**Review by Gudahtt — DISMISSED** [review]
LGTM!

File: package.json (hunk 1 above)
Comment (brad-decker) [reply package.json:562]: The only reason I did that is it was the version that installed for me when I installed 18 via nvm and it matches the node browsers executor in circleci. I am happy to make it >= 18?

Comment (brad-decker) [reply package.json:562]: would it be `>= 18 || >= 10' ?

Comment (brad-decker) [reply package.json]: pushed with '>= 18' but happy to revise

Comment (Gudahtt) [reply package.json]: Sounds good! The `^18 || ^20` version was for if we wanted to exclude the non-LTS releases, but I think `>=18` makes sense.

**Comment by brad-decker** [comment]
@Gudahtt looks like a ganache issue? https://github.com/trufflesuite/ganache/issues/1503

**Comment by brad-decker** [comment]
Upgrading to ganache 7.9 locally works to resolve the error: 

```shell
metamask-extension on  use-node-18 [$] via ⬢ v18.17.0 took 5.6s 
➜ yarn ganache
bigint: Failed to load bindings, pure JS will be used (try npm run rebuild?)
This version of µWS is not compatible with your Node.js build:

Error: Cannot find module './uws_darwin_arm64_108.node'
Falling back to a NodeJS implementation; performance may be degraded.


ganache v7.0.4 (@ganache/cli: 0.1.5, @ganache/core: 0.1.5)
Starting RPC server
```

after upgrade:

```shell
➜ yarn ganache   
bigint: Failed to load bindings, pure JS will be used (try npm run rebuild?)
ganache v7.9.0 (@ganache/cli: 0.10.0, @ganache/core: 0.10.0)
Starting RPC server
```

**Comment by brad-decker** [comment]
Kind of a large upgrade, though, probably should try that on a new branch and see if it breaks CI in 16 before conflicting that with a node 18 update.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**